packages/jsii-pacmak/lib/targets/java.ts

```diff
@@ -531,7 +533,7 @@ class JavaGenerator extends Generator {
         this.code.line(`${access} final static ${propType} ${propName};`);
     }
 
-    private emitProperty(cls: spec.Type, prop: spec.Property, includeGetter = true, overrides = false) {
+    private emitProperty(cls: spec.Type, prop: spec.Property, includeGetter = true, overrides: boolean = !!prop.overrides) {
```

Comment:
If you can obtain this from `prop`, why do you need it in the signature? Also I am shocked that you can refer to another argument here...

Reply:
So I can obtain it from `prop` but in the case where we generate the proxy class for an interface, I want to output `@Override` for everything (since it's supposed to implement the interface). In this case, the `prop` won't tell me what to do.

And yes, you can refer to another argument, because the default value for JS arguments is syntax sugar for prefixing the code block with `if (arg == null) { arg = default; }`. So other arguments are known (but you sometimes need be careful that default assignments are de-sugared in declaration order).